### DISCUSSION TITLE
Make RC_MAX_LAYERS  and RC_MAX_NEIS configurable at build time

### DISCUSSION THREAD

**Discussion by ghost** [recastnavigation/recastnavigation#605] [Ideas]
[Those](https://github.com/recastnavigation/recastnavigation/blob/b43c82c8087f7e277077dcb7cb1e6c626d5550ce/Recast/Source/RecastLayers.cpp#L29-L32) are currently global constants, which require modifying recast's source itself (see [this issue](https://github.com/Unvanquished/Unvanquished/issues/2403) from a game that uses recast).

Would it be ok to make it possible to make this a cmake option instead? This way, it would be possible for users to at least change the value at compilation time without having to modify recast's code directly.
If yes, I'll write a PR for this.

Maybe it would be possible to make the value dynamic at runtime too, but I'm not sure how much this would require patching or if it would be a bad idea for performance or other techical reasons.

**Comment by grahamboree** [comment]
Hi!  I'm not terribly familiar with cmake's build configuration system, but I assume it's essentially setting C preprocessor variables like other build systems do.

I'd be open to moving these integers to `#define`'s instead since it'd expose configuration to build systems like cmake, assuming that's what cmake options are operating with.

**Reply by ghost** [reply]
yes, cmake can expose variables like any other build system. It's rather simple to do, too, and recast already does that: https://github.com/recastnavigation/recastnavigation/blob/main/version.h.in is used as source by cmake to generate the final header: https://github.com/recastnavigation/recastnavigation/blob/main/CMakeLists.txt#L23-L25

Of course, I do not think this value should go in a version.h file, but a configure.h one, why not? It depends on Recast's codestyle.
